SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -632,6 +616,26 @@ void FSpatialNetGUIDCache::RemoveEntityNetGUID(Worker_EntityId EntityId)
 	{
 		UnrealObjectRefToNetGUID.Remove(StablyNamedRefOption.GetValue());
 	}
+
+	// Remove dynamic subobjects.
+	if (USpatialActorChannel* Channel = SpatialNetDriver->GetActorChannelByEntityId(EntityId))
+	{
+		for (auto& ObjReplicatorPair : Channel->ReplicationMap)
+		{
+			TWeakObjectPtr<UObject> WeakObjectPtr = ObjReplicatorPair.Value->GetWeakObjectPtr();
+			if (WeakObjectPtr.IsValid())
+			{
+				if (FNetworkGUID* ObjectNetGUID = NetGUIDLookup.Find(WeakObjectPtr))
+				{
+					if (FUnrealObjectRef* SubobjectRef = NetGUIDToUnrealObjectRef.Find(*ObjectNetGUID))
+					{
+						UnrealObjectRefToNetGUID.Remove(*SubobjectRef);
+						NetGUIDToUnrealObjectRef.Remove(*ObjectNetGUID);
+					}
+				}
+			}
+		}
+	}
```

Comment:
Was this moved after deliberately? I think I remember seeing a PR that moved this around to fix another bug.

Reply:
I moved it after as it meant didn't need to gate actor removal with a conditional (as this would remove an actor that had been replicated), but if we think it might break things we should keep the ordering. Moved it back to what it was before

Reply:
nevermind, I was thinking about something else, will see again if this should matter

Reply:
hmm, I don't think it should but isn't a big deal, will leave the reordering I've now done as it decreases risk